### DISCUSSION TITLE
How to force and raise an Propertychanged for styledproperty in avalonia templated control

### DISCUSSION THREAD

**Discussion by CastelloBrancoTecnologia** [AvaloniaUI/Avalonia#12767] [Q&A] [answered]
I have an templated control that is not updating some properties in Binding (PAth1, Path2, etc)

How to generate PropertyChnaged ?

the code is 

public partial class RatingControl : ContentControl
{
    public RatingControl()
    {
    }

    public static readonly StyledProperty<int> EstrelasProperty = AvaloniaProperty.Register<RatingControl, int>(nameof(Estrelas), 0);

    public int Estrelas
    {
        get => GetValue(EstrelasProperty);
        set
        {
            SetValue<int>(EstrelasProperty, value);

            Adjust();
        }
    }

    public static readonly StyledProperty<string> Path1Property = AvaloniaProperty.Register<RatingControl, string>(nameof(Path1), "");
    public static readonly StyledProperty<string> Path2Property = AvaloniaProperty.Register<RatingControl, string>(nameof(Path2), "");
    public static readonly StyledProperty<string> Path3Property = AvaloniaProperty.Register<RatingControl, string>(nameof(Path3), "");
    public static readonly StyledProperty<string> Path4Property = AvaloniaProperty.Register<RatingControl, string>(nameof(Path4), "");
    public static readonly StyledProperty<string> Path5Property = AvaloniaProperty.Register<RatingControl, string>(nameof(Path5), "");

    public string Path1 { get => GetValue(Path1Property); set => SetValue<string>(Path1Property, value); }
    public string Path2 { get => GetValue(Path2Property); set => SetValue<string>(Path2Property, value); }
    public string Path3 { get => GetValue(Path3Property); set => SetValue<string>(Path3Property, value); }
    public string Path4 { get => GetValue(Path4Property); set => SetValue<string>(Path4Property, value); }
    public string Path5 { get => GetValue(Path5Property); set => SetValue<string>(Path5Property, value); }

    public void Adjust ()
    {
        Path1 = Estrelas >= 1 ? PathEstrelaAcesa : PathEstrelaApagada;
        Path2 = Estrelas >= 2 ? PathEstrelaAcesa : PathEstrelaApagada;
        Path3 = Estrelas >= 3 ? PathEstrelaAcesa : PathEstrelaApagada;
        Path4 = Estrelas >= 4 ? PathEstrelaAcesa : PathEstrelaApagada;
        Path5 = Estrelas >= 5 ? PathEstrelaAcesa : PathEstrelaApagada;

        **//  <=========== HOW Raise Propertychanged for Path1, Path2, Path3 and Path4 ????????

       // BECAUSE SVG CONTROL IS NOT UPDATING BINDED PATH IN XAML**
    }

    public static readonly StyledProperty<string> PathEstrelaAcesaProperty = AvaloniaProperty.Register<RatingControl, string>(nameof(PathEstrelaAcesa), "");

    public string PathEstrelaAcesa
    {
        get => GetValue(PathEstrelaAcesaProperty);
        set
        {
            SetValue<string>(PathEstrelaAcesaProperty, value);

            Adjust();
        }
    }

    public static readonly StyledProperty<string> PathEstrelaApagadaProperty = AvaloniaProperty.Register<RatingControl, string>(nameof(PathEstrelaApagada), "");

    public string PathEstrelaApagada
    {
        get => GetValue(PathEstrelaApagadaProperty);
        set
        {
            SetValue<string>(PathEstrelaApagadaProperty, value);

            Adjust();
        }
    }

==================================================================

XAML

<Styles
    xmlns="https://github.com/avaloniaui"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:cc="clr-namespace:CBT.RatingControl.Avalonia"
    xmlns:sys="using:System">
    <Style Selector="cc|RatingControl">
        <Setter Property="Template">
            <ControlTemplate>
            ..... 

           ****

            <Svg Path="{Binding Path1, RelativeSource={RelativeSource TemplatedParent}}" />
            <Svg Path="{Binding Path2, RelativeSource={RelativeSource TemplatedParent}}" />
            <Svg Path="{Binding Path3, RelativeSource={RelativeSource TemplatedParent}}" />
            <Svg Path="{Binding Path4, RelativeSource={RelativeSource TemplatedParent}}" />
            <Svg Path="{Binding Path5, RelativeSource={RelativeSource TemplatedParent}}" />

            .....
            </ControlTemplate>
        </Setter>
    </Style>
</Styles>

**Comment by timunie** [comment, marked as answer]
Better to override OnPropertyChanged and filter for e.Property. example: 

https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src%2FAvalonia.Samples%2FCustomControls%2FRatingControlSample
